txn_writer_test.go

```diff
@@ -567,3 +570,35 @@ func TestSafeWriter_VendorDotGitPreservedWithForceVendor(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestCalculatePrune(t *testing.T) {
+	h := test.NewHelper(t)
+	defer h.Cleanup()
+
+	vendorDir := "vendor"
+	h.TempDir(vendorDir)
+	h.TempDir(filepath.Join(vendorDir, "github.com/keep/pkg/sub"))
+	h.TempDir(filepath.Join(vendorDir, "github.com/prune/pkg/sub"))
+
+	toKeep := []string{
+		filepath.FromSlash("github.com/keep/pkg"),
+		filepath.FromSlash("github.com/keep/pkg/sub"),
+	}
+
+	got, err := calculatePrune(h.Path(vendorDir), toKeep, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	sort.Sort(byLen(got))
+
+	want := []string{
+		h.Path(filepath.Join(vendorDir, "github.com/prune/pkg/sub")),
+		h.Path(filepath.Join(vendorDir, "github.com/prune/pkg")),
+		h.Path(filepath.Join(vendorDir, "github.com/prune")),
+	}
+
+	if !reflect.DeepEqual(want, got) {
+		t.Fatalf("expected %s, got %s", want, got)
```

Comment:
I've removed the tabs since they are not needed here. An error would like this now:

```
--- FAIL: TestCalculatePrune (0.00s)
	txn_writer_test.go:602: calculated prune paths are not as expected.
		(WNT) [/private/var/folders/hb/pdqmtgn954zbj319w8djtncr0000gn/T/gotest236842704/vendor/github.com/prune/pkg/sub /private/var/folders/hb/pdqmtgn954zbj319w8djtncr0000gn/T/gotest236842704/vendor/github.com/prune/pkg]
		(GOT) [/private/var/folders/hb/pdqmtgn954zbj319w8djtncr0000gn/T/gotest236842704/vendor/github.com/prune/pkg/sub /private/var/folders/hb/pdqmtgn954zbj319w8djtncr0000gn/T/gotest236842704/vendor/github.com/prune/pkg /private/var/folders/hb/pdqmtgn954zbj319w8djtncr0000gn/T/gotest236842704/vendor/github.com/prune]
```